research-app/src/CatalogItem.vue

```diff
@@ -0,0 +1,223 @@
+      <span id="buttons-container">
+        <a href="#" v-hide="!hasFocus" @click="handleDelete" class="icon-link"><font-awesome-icon class="icon" icon="times"/></a>
+        <a href="#" v-hide="!hasFocus" @click="handleToggle" class="icon-link"><font-awesome-icon v-if="visible" class="icon" icon="eye"/><font-awesome-icon v-if="!visible" class="icon" icon="eye-slash"/></a>
+      </span>
+    </div>
+    <transition-expand>
+      <div v-if="isSelected" class="detail-container">
+        <div class="detail-row"><span class="prompt">URL:</span><span class="url-holder">{{ catalog.url }}</span></div>
+        <div class="detail-row" v-if="catalog.description.length > 0"><span class="prompt">Description:</span><span>{{ catalog.description }}</span></div>
+        <div class="detail-row"><span class="prompt">Color:</span>
+          <v-popover class="circle-popover">
+            <font-awesome-icon icon="circle" size="lg" :style="{ 'color' : `rgba(${this.color.r}, ${this.color.g}, ${this.color.b}, ${this.color.a})` }"></font-awesome-icon>
+            <template slot="popover">
+              <vue-color-chrome
+              :value="this.color"
+              @input="handleColorChange"
+              ></vue-color-chrome>
+            </template>
+          </v-popover>
+        </div>
+      </div>
+    </transition-expand>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from "vue-property-decorator";
+import { ImagesetInfo } from "@wwtelescope/engine-vuex";
+import { VNode, VNodeDirective } from 'vue';
+import { Color } from '@wwtelescope/engine';
+import { mapGetters, mapMutations } from "vuex";
+
+/** v-hide directive take from https://www.ryansouthgate.com/2020/01/30/vue-js-v-hide-element-whilst-keeping-occupied-space/ */
+// Extract the function out, up here, so I'm not writing it twice
+const update = (el: HTMLElement,
+    binding: VNodeDirective,
+    _vnode: VNode,
+    _oldVnode: VNode) => el.style.visibility = (binding.value) ? "hidden" : "";
+
+/**
+ * Hides an HTML element, keeping the space it would have used if it were visible (css: Visibility)
+ */
+Vue.directive("hide", {
+    // Run on initialisation (first render) of the directive on the element
+    bind: update,
+    // Run on subsequent updates to the value supplied to the directive
+    update: update
+})
+
+interface VueColorData {
+  'rgba': {
+    'a': number;
+    'r': number;
+    'g': number;
+    'b': number;
+  };
+}
+
+@Component
+export default class CatalogItem extends Vue {
+    @Prop({required: true}) catalog!: ImagesetInfo;
+    @Prop({required: false, default: Color.fromArgb(1, 255, 255, 255)}) defaultColor!: Color;
+    hasFocus = false;
+    isSelected = false;
+    visible = true;
+    color = new Color();
+    
+    beforeCreate(): void {
+      const wwtEngineNamespace = "wwt-engine";
+      const wwtResearchAppNamespace = "wwt-research";
```

Comment:
I see these constants have been duplicated a few times here and there. It might make sense to centralize them in a module somewhere, and import them.

Reply:
I agree. My initial thought was to put them in `main.ts`, but that would require the components to import from `main.ts` and create a circular dependency as far as imports go. I've moved these to `namespaces.ts` in `research-app/src`, but if we end up using a similar structure elsewhere, we can/should move it further up the repository tree.